### PR TITLE
fix(container): fix list ForEach bug and add comprehensive tests

### PR DESCRIPTION
**Pull request by amery**
## Summary

This PR fixes a critical bug in the container/list package and adds comprehensive test coverage for the container packages.

### Bug Fix
- **container/list**: Fixed ForEach method that was only iterating over the first element due to inverted callback return semantics

### Test Coverage
- **container/list**: Added comprehensive unit tests and example tests
- **container/set**: Added full test coverage including edge cases and thread safety tests  
- **container/slices**: Added tests for CustomSet methods including nil receivers and concurrent access

### Documentation
- **container/README.md**: Updated with accurate examples showing ForEach behavior and enhanced usage examples

### Minor Fixes
- **web/forms, web/resource**: Added shellcheck disable directives for local variable usage

## Test Results

All tests pass with the container package now at 32.9% coverage (up from near 0% for set and slices packages).

```
[3/9] Testing module: container    ✓ (container)          32.9% of statements in ./...
```

___

### **PR Type**
Bug fix, Tests, Documentation

___

### **Description**
- Fixed critical `ForEach` bug in container/list package

- Added comprehensive test coverage for all container packages

- Updated documentation with accurate examples and usage

- Minor shellcheck fixes for web scripts

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["Bug in ForEach"] --> B["Fixed iteration logic"]
  B --> C["Added comprehensive tests"]
  C --> D["Updated documentation"]
  E["Shellcheck warnings"] --> F["Added disable directives"]
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>list.go</strong><dd><code>Fix ForEach iteration bug</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/230/files#diff-486b241b96b7b1972533926b248e8a5d4e06c055c4cb3c9c4f38c4629e93e27d">+6/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>8 files</summary><table>
<tr>
  <td><strong>list_doc_test.go</strong><dd><code>Add comprehensive example tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/230/files#diff-298acb3b7034a8db928e449a9d81261cc65d7d40f69e9e61ad5b81dc6833179d">+120/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>list_more_test.go</strong><dd><code>Add extensive unit tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/230/files#diff-50197182181aa29f46b0d14721c8f903abaf54c8a3a2edc0d0003b2dab38c853">+390/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>list_test.go</strong><dd><code>Add ForEach bug regression tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/230/files#diff-033633f055a67b6a3a5950d83b816a000e73a90324065a40afa2ee529837a237">+176/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>example_test.go</strong><dd><code>Add set package example tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/230/files#diff-667e9e3e9b2dc1a8ef125ebb5375b6df789f69c4d8f422b522ec9db971cf7c5d">+218/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>set_simple_test.go</strong><dd><code>Add basic set operation tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/230/files#diff-f60b77c8fa54caae9aab53d9f24fa9fe6c24b9166fadbf9594d2127051fbcf39">+53/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>set_test.go</strong><dd><code>Add comprehensive set tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/230/files#diff-4673c763c17ba2e98c38cb15a4cc0226254fc9e854bfb5de5c6578ee95b0b02e">+862/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>example_test.go</strong><dd><code>Add slices package example tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/230/files#diff-77a74ce0ae6306cf77b75763048c771ea429a55fe8a8b898d0ddebe017d5368d">+191/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>set_fn_test.go</strong><dd><code>Add CustomSet comprehensive tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/230/files#diff-83fe116481765747a055b1a75ae142af41c02dd786bd0a9ddf4581125e5f54dd">+397/-0</a>&nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Miscellaneous</strong></td><td><details><summary>3 files</summary><table>
<tr>
  <td><strong>strconv_gen.sh</strong><dd><code>Add shellcheck disable directive</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/230/files#diff-feaa1f3baa038d9a1c9b07e676e8db6bd8906539b932cdb10ec4a56179771803">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>value_gen.sh</strong><dd><code>Add shellcheck disable directive</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/230/files#diff-c045caeee6897b247668a7c0081fdbd359aaa0e6266601f7a2e86e709f6e47d1">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>resolve_gen.sh</strong><dd><code>Add shellcheck disable directive</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/230/files#diff-6a551181f1bb339e329626e5d8ee10c10ae8383f431584e33f2f180ff8cea87b">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Documentation</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>README.md</strong><dd><code>Update documentation with examples</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/darvaza-proxy/x/pull/230/files#diff-869b8bb231f558015e37f0ee96ddd16a4580d3502d80a97154556641acbafd47">+25/-1</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

</details>

___

## Summary by CodeRabbit

* **Documentation**
  * Enhanced usage examples and clarifications in the container package documentation, including new examples for list, set, and slices features.
  * Updated documentation to note comprehensive test coverage with examples.

* **Tests**
  * Added extensive example and unit tests for list, set, and slices data structures, covering creation, operations, iteration, cloning, filtering, and error handling.
  * Improved test coverage for both primitive and custom types, including thread safety and edge cases.
  * Introduced benchmarks and concurrency tests for set operations.

* **Style**
  * Added directives in shell scripts to suppress specific shellcheck warnings, with no changes to script behavior.

### CODE REVIEW

File: container/set/set_simple_test.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	if err != nil {
+		t.Fatalf("New failed: %v", err)
+	}
+
+	doTestPush(t, s)
+	doTestContains(t, s)
+	doTestGet(t, s)
+	doTestPop(t, s)
+	doTestVerifyRemoved(t, s)
+}
```
[the remark this answers is not in the document]
Comment (amery) [reply container/set/set_simple_test.go:55]: this should be sorted now
